File: rust/src/models/capacity_pricing.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct CapacityPricing {
    pub included: i32,

    pub overage_rate: rust_decimal::Decimal,

    pub rate: rust_decimal::Decimal,
}

impl CapacityPricing {
    pub fn new(
        included: i32,
        overage_rate: rust_decimal::Decimal,
        rate: rust_decimal::Decimal,
    ) -> Self {
        Self {
            included,
            overage_rate,
            rate,
        }
    }
}

File: typescript/src/models/capacityPricing.ts
// this file is @generated

export interface CapacityPricing {
  included: number;

  overageRate: string;

  rate: string;
}

export const CapacityPricingSerializer = {
  _fromJsonObject(object: any): CapacityPricing {
    return {
      included: object["included"],
      overageRate: object["overage_rate"],
      rate: object["rate"],
    };
  },

  _toJsonObject(self: CapacityPricing): any {
    return {
      included: self.included,
      overage_rate: self.overageRate,
      rate: self.rate,
    };
  },
};
